### PR TITLE
MBS-11598 Fix test freezes

### PR DESCRIPTION
**Pull request by RuslanMingaliev**

### CODE REVIEW

File: subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/Intents.kt [hunk 1]
```diff
@@ -34,6 +37,21 @@ object Intents {
     private fun Instrumentation.ActivityResult.toReadableString() =
         "ActivityResult { resultCode=${this.resultCode} resultData=${this.resultData} }"
 
+    /**
+     * TODO: Remove after MBS-11523
+     * Test will freeze in the end if instrumentation intents are stubbed.
+     * {@link androidx.test.core.app.InstrumentationActivityInvoker#finishActivity finishActivity}
+     */
+    class NotInstrumentationIntentMatcher : TypeSafeMatcher<Intent>() {
+        override fun describeTo(description: Description?) {
```
Comment (sergioboi) [inline subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/Intents.kt]: description could be safely made non nullable

File: subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/Intents.kt [hunk 2]
```diff
@@ -34,6 +37,21 @@ object Intents {
...
+            description?.appendText("not instrumentation intent")
+        }
+
+        override fun matchesSafely(item: Intent?): Boolean {
```
Comment (sergioboi) [inline subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/Intents.kt]: the same

File: subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/Intents.kt [hunk 3]
```diff
@@ -34,6 +37,21 @@ object Intents {
...
+            return item?.hasInstrumentationActivityComponent()?.not() ?: true
```
Comment (sergioboi) [inline subprojects/android-test/ui-testing-core/src/main/kotlin/com/avito/android/test/Intents.kt]: The mother of GOD

File: subprojects/android-test/ui-testing-core/src/main/kotlin/androidx/test/core/app/InstrumentationActivityInvokerExtension.kt [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+/**
+ * TODO: Remove after MBS-11523
+ */
+fun Intent.hasInstrumentationActivityComponent(): Boolean {
+    val instrumentationComponents: List<String> = listOf(
+        InstrumentationActivityInvoker.EmptyActivity::class.java,
+        InstrumentationActivityInvoker.EmptyFloatingActivity::class.java,
+        InstrumentationActivityInvoker.BootstrapActivity::class.java
+    ).map { it.name }
```
Comment (sergioboi) [inline subprojects/android-test/ui-testing-core/src/main/kotlin/androidx/test/core/app/InstrumentationActivityInvokerExtension.kt]: jfyi
you could delete the `map` and inline `.name`

File: subprojects/android-test/ui-testing-core/src/main/kotlin/androidx/test/core/app/InstrumentationActivityInvokerExtension.kt [hunk 5]
```diff
@@ -0,0 +1,16 @@
+package androidx.test.core.app
+
+import android.content.Intent
+
+/**
+ * TODO: Remove after MBS-11523
+ */
+fun Intent.hasInstrumentationActivityComponent(): Boolean {
```
Comment (eugene-krivobokov) [inline subprojects/android-test/ui-testing-core/src/main/kotlin/androidx/test/core/app/InstrumentationActivityInvokerExtension.kt]: Can it be internal?

Comment (RuslanMingaliev) [reply subprojects/android-test/ui-testing-core/src/main/kotlin/androidx/test/core/app/InstrumentationActivityInvokerExtension.kt]: Yes, it has to be internal

**Review by sergioboi — APPROVED** [review]

**Pull request merged**